### ISSUE TITLE
Combination `.mend(???).within(???)` not working inside `split` block

### ISSUE THREAD

**Issue by NPCRUS** [propensive/contingency#19]
```scala
trait Defooler[T] {
    def defool(a: String): Option[String]
  }
  object Defooler extends Derivation[Defooler] {
    inline def join[DerivationType <: Product](using ProductReflection[DerivationType]): Defooler[DerivationType] = ???

    inline def split[DerivationType](using SumReflection[DerivationType]): Defooler[DerivationType] = str =>
      mend {
        case e: VariantError =>
          Left(e.getMessage)
      }.within {
        delegate(str)([VariantType <: DerivationType] => defooler =>
          defooler.defool(str)  
        )
      }
  }
```
produces:
```
value within is not a member of Any.
An extension method was tried, but could not be fully constructed:

    contingency.within()

    failed with:

        value within: <overloaded contingency.within> does not take parameters
``` 
at `.within`

**Comment by propensive** [comment]
It was straightforward enough to move it with the "transfer" button!

**Comment by propensive** [comment]
The problem is that the transparent inline method, `mend` (and its siblings, `tend` and so on), has a declared return type of `Any` but should always refine its type to some parameterization of `Mend` based on the `case`s that are declared. The extension method `within` would then applicable.

But inside an `inline` block, the surrounding `inline` is expanded before the inner `mend` is, so we know nothing more about the return type of `mend { cases }` at the point the compiler tries to find the right `within` extension method.

We could redefine the return type of `mend` to something fixed, but more precise. That would help to get `within` selected. But the `mendWithin` implementation of `within` needs more precise details of the type generated by the `mend` macro, just to typecheck its body.

We might be able to regenerate those type details in the `mendWithin` macro, but by then I think it's too late. The compiler would need to typecheck `within`'s body before this, and that shouldn't work without knowing what cases are being handled.

In short, I'm not hopeful this can be fixed without some drastic changes.
